lib/improver/utilities/cube_units.py

```diff
@@ -32,14 +32,126 @@
 
 import numpy as np
 
+import iris
 from iris.exceptions import CoordinateNotFoundError
 
 from improver.units import DEFAULT_UNITS
 
 
-def enforce_coordinate_units_and_dtypes(cubes, coordinates, inplace=True):
+def enforce_units_and_dtypes(cubes, coords=None, enforce=True):
+    """
+    Function to check the units and datatypes of cube diagnostics and
+    coordinates against the manifest in improver.units, with option to
+    enforce or fail for non-conforming data.
+
+    Args:
+        cubes (iris.cube.Cube or iris.cube.CubeList):
+            Cube or list of cubes to be checked
+
+    Kwargs:
+        coords (list or None):
+            List of coordinate names to check.  If None, checks all
+            coordinates present on the input cubes.
+        enforce (bool):
+            If True, this function returns a list of conformant cubes.
+            If False, a ValueError is thrown if the cubes do not conform.
+
+    Raises:
+        ValueError: if "enforce=False" and the input cubes do not conform
+            to the datatypes and units standard.
+
+    Returns:
+        new_cubes (iris.cube.CubeList):
+            New cubelist with conformant datatypes and units
+    """
+    # convert input to CubeList
+    if isinstance(cubes, iris.cube.Cube):
+        cubes = [cubes]
+
+    # return a new cube list from the inputs in which data units and datatypes
+    # have been enforced
+    new_cubes = _enforce_diagnostic_units_and_dtypes(cubes, inplace=False)
```

Comment:
Any idea on whether this could be a significant time sink? By performing the conversion, regardless of whether we want to enforce the change, we will always perform the calculation on every cube load in the suite. Whilst this is not likely to be a huge calculation each time, multiplied many times it is more than nil. We have the units specified in the cubes, so if enforce=False, would it not make sense to simply check that they match the manifest?

If you agree that this might be sensible, and you agree to the refactoring below that I propose (to consolidate the two functions), the comparison could be moved into that function. All this is up for discussion.